### DISCUSSION TITLE
Accessing q:Q object in a wave app on startup

### DISCUSSION THREAD

**Discussion by n-srinidhi** [h2oai/wave#1777] [Q&A] [answered]
How to access the q:Q object in the `'on_startup'` function while running a Wave App?
Basically, I am trying to create a background task, which will execute periodically to update the stats cards and other components in my Wave App. Like this:

```
logging.info('Starting background thread..')
    while True:
        with concurrent.futures.ThreadPoolExecutor() as pool:
            frame=await q.exec(pool, load_data)            
            print(frame)
            time.sleep(1)
```

when I try to attach this to `@app('/')`, It requires the q:Q object to be passed as an argument.

**Comment by mturoci** [comment, marked as answer]
The q object is only accesible when the client (browser) is connected - sends data to Wave app server. `on_startup` is a hook for when the Wave app server is booted up (no client connected yet).

Your 2 options in this case are:
* Start the background job in app/client init within the serve function.
* Start the background job in the on startup and pass the q object for subsequent updates afterwards - slightly more complex.
